### PR TITLE
CI: split x86_64-msvc-ext job

### PR DESCRIPTION
**Pull request by marcoieni**
I'm splitting this job into two so that we can use free runners instead of large runners.

try-job: x86_64-msvc-ext1
try-job: x86_64-msvc-ext3

### CODE REVIEW

**Comment by MarcoIeni** [comment]
trying again to check if there are cache misses somewhere.

**Comment by ChrisDenton** [comment]
Oh btw I would like to rejoin it with msvc-ext2. Splitting them was only intended to be a temporary hack. If we do want them split long term then there's hopefully a more principled way to do it.

**Comment by MarcoIeni** [comment]
I have a question Chris, I see that the following env variables are different:

```
      HOST_TARGET: x86_64-pc-windows-msvc
      RUST_CONFIGURE_ARGS: --build=x86_64-pc-windows-msvc --enable-lld --save-toolstates=/tmp/toolstate/toolstates.json
      DEPLOY_TOOLSTATES_JSON: toolstates-windows.json
``` 

should the second job have the same env variables as the first one?

**Comment by MarcoIeni** [comment]
For the reviewer of this PR: I tried splitting the job because it took too long: [2h 52m](https://github.com/rust-lang-ci/rust/actions/runs/11938781386/usage)

So I introduced a third job. Let's see how it goes!

**Comment by MarcoIeni** [comment]
Nice, the timings look reasonable!
image

**Comment by ChrisDenton** [comment]
> I have a question Chris, I see that the following env variables are different:
> 
> ```
>       HOST_TARGET: x86_64-pc-windows-msvc
>       RUST_CONFIGURE_ARGS: --build=x86_64-pc-windows-msvc --enable-lld --save-toolstates=/tmp/toolstate/toolstates.json
>       DEPLOY_TOOLSTATES_JSON: toolstates-windows.json
> ``` 
> 
> should the second job have the same env variables as the first one? If not, why?

I can't recall exactly but probably because the second job doesn't update the toolstate. It basically just very arbitrarily splits off a part of checktools.sh.

File: src/ci/github-actions/jobs.yml [hunk 1]
```diff
@@ -406,6 +405,14 @@ auto:
       RUST_CONFIGURE_ARGS: --build=x86_64-pc-windows-msvc --enable-lld
     <<: *job-windows
 
+  - image: x86_64-msvc-ext3
```
Comment (Kobzol) [inline src/ci/github-actions/jobs.yml:408]: Could you please add a comment here explaining why are the jobs split, and what is this specific job doing (checking tools and uploading toolstate)? Our jobs are critically undocumented, so if we're making changes, we might as well leave it in a better state than before.

Comment (MarcoIeni) [reply src/ci/github-actions/jobs.yml:409]: Done 👍
I'm retrying bors because I removed an env variable from the first job, too.

**Pull request merged**